### PR TITLE
Chore(metrics): add new timer for full preview load time tracking

### PR DESCRIPTION
**Pull request by JustinHoldstock**
This puts our apdex and preview load timings at parity with our legacy metrics system.

TODO:
- [x] tests

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @JustinHoldstock has signed the CLA. Thanks for the pull request!

File: src/lib/Preview.js [hunk 1]
```diff
@@ -1245,8 +1250,10 @@ class Preview extends EventEmitter {
      */
     finishLoading(data = {}) {
         if (this.file && this.file.id) {
-            const tag = Timer.createTag(this.file.id, LOAD_METRIC.fullDocumentLoadTime);
-            Timer.stop(tag);
+            const contentLoadTag = Timer.createTag(this.file.id, LOAD_METRIC.fullDocumentLoadTime);
```
Comment (jeremypress) [inline src/lib/Preview.js]: Should this move to finshLoading of docBase?

File: src/lib/Preview.js [hunk 2]
```diff
@@ -1530,34 +1537,34 @@ class Preview extends EventEmitter {
...
+            [LOAD_METRIC.convertTime]: convertTime.elapsed || 0,
+            [LOAD_METRIC.downloadResponseTime]: downloadTime.elapsed || 0,
+            [LOAD_METRIC.fullDocumentLoadTime]: contentLoadTime.elapsed || 0,
             ...this.createLogEvent()
         };
 
         this.emit(PREVIEW_METRIC, event);
 
-        Timer.reset([infoTag, convertTag, downloadTag, fullLoadTag]);
+        Timer.reset([infoTag, convertTag, downloadTag, contentLoadTag, previewLoadTag]);
```
Comment (jeremypress) [inline src/lib/Preview.js]: why not just rename the timer fullDocumentLoadTime to contentLoadTime?

File: src/lib/Preview.js [hunk 3]
```diff
@@ -753,6 +754,10 @@ class Preview extends EventEmitter {
             );
         }
 
+        // Start the preview duration timer when the user starts to perceive preview's load
+        const tag = Timer.createTag(this.file.id, PREVIEW_LOAD_TIME_TAG);
```
Comment (ConradJChan) [inline src/lib/Preview.js]: Do we care about this metric if `reload` is called with `skipServerUpdate = true`? That code path seems to bypass `load`

File: src/lib/Preview.js (hunk 1 above)
Comment (JustinHoldstock) [reply src/lib/Preview.js]: We want to track load times across all file types, not just documents

File: src/lib/Preview.js (hunk 2 above)
Comment (JustinHoldstock) [reply src/lib/Preview.js]: That would be a lot more consistent. Can do!

File: src/lib/Preview.js (hunk 3 above)
Comment (JustinHoldstock) [reply src/lib/Preview.js]: For us, our most important metrics are going to be related to our network response times, and served content times. If it's coming from a "reload" most of the content will already be in the browser cache, which we don't care so much about.

File: src/lib/Preview.js (hunk 1 above)
Comment (jeremypress) [reply src/lib/Preview.js]: sounds good after rename

File: src/lib/Preview.js [hunk 4]
```diff
@@ -1530,34 +1537,34 @@ class Preview extends EventEmitter {
...
-        const convertTag = Timer.createTag(this.file.id, LOAD_METRIC.convertTime);
-        const downloadTag = Timer.createTag(this.file.id, LOAD_METRIC.downloadResponseTime);
-        const fullLoadTag = Timer.createTag(this.file.id, LOAD_METRIC.fullDocumentLoadTime);
+        const { id } = this.file;
+
+        const infoTag = Timer.createTag(id, LOAD_METRIC.fileInfoTime);
+        const convertTag = Timer.createTag(id, LOAD_METRIC.convertTime);
+        const downloadTag = Timer.createTag(id, LOAD_METRIC.downloadResponseTime);
+        const contentLoadTag = Timer.createTag(id, LOAD_METRIC.contentLoadTime);
+        const previewLoadTag = Timer.createTag(id, PREVIEW_LOAD_TIME_TAG);
```
Comment (jstoffan) [inline src/lib/Preview.js]: Seems like this tag should use the same convention as the other values. Something like `LOAD_METRIC.previewLoadTime`.

**Pull request merged**